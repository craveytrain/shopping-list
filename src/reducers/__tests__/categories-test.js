import reducer from 'reducers/categories';
import {
  addCategory,
  removeCategory,
  nameCategory
} from 'actions/categories';
import deepFreeze from 'deep-freeze';

describe('categories reducer', () => {
  describe('addCategory', () => {
    it('should add a category to the list', () => {
      const nextState = reducer(
        {},
        addCategory({
          name: 'Meat'
        })
      );

      expect(nextState).toMatchSnapshot();
    });

    it('should add a category to the list with an id', () => {
      const nextState = reducer(
        {},
        addCategory({
          name: 'Meat',
          id: 'meat'
        })
      );

      expect(nextState).toMatchSnapshot();
    });

    it('should add a category to the list by string', () => {
      const nextState = reducer(
        {},
        addCategory('Meat')
      );

      expect(nextState).toMatchSnapshot();
    });
  });

  describe('removeCategory', () => {
    it('should remove a category from the list', () => {
      const initialState = {
        byId: {
          dairy: {
            name: 'Dairy'
          },
          meat: {
            name: 'Meat'
          }
        },
        allIds: [ 'dairy', 'meat' ]
      };

      deepFreeze(initialState);

      const nextState = reducer(initialState, removeCategory('dairy'));

      expect(nextState).toMatchSnapshot();
    });
  });

  describe('nameCategory', () => {
    it('should set the name of the selected category from the list', () => {
      const initialState = {
        byId: {
          dairy: {
            name: 'Dairy'
          },
          meat: {
            name: 'Meat'
          }
        },
        allIds: [ 'dairy', 'meat' ]
      };

      deepFreeze(initialState);

      const nextState = reducer(
        initialState,
        nameCategory('dairy', 'Milk & Cheese')
      );

      expect(nextState).toMatchSnapshot();
    });
  });

  it('can be used with reduce', () => {
    const actionQueue = [
      addCategory({
        id: 'dairy',
        name: 'Dairy'
      }),
      addCategory({
        name: 'Produce',
        id: 'produce'
      }),
      addCategory({
        name: 'Meat'
      }),
      nameCategory('dairy', 'Milk & Cheese'),
      removeCategory('produce')
    ];

    const finalState = actionQueue.reduce(reducer, {});

    expect(finalState).toMatchSnapshot();
  });
});
