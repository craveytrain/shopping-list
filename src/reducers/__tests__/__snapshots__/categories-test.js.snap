// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories reducer addCategory should add a category to the list 1`] = `
Object {
  "allIds": Array [
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
  ],
  "byId": Object {
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015": Object {
      "id": "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
      "name": "Meat",
    },
  },
}
`;

exports[`categories reducer addCategory should add a category to the list by string 1`] = `
Object {
  "allIds": Array [
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
  ],
  "byId": Object {
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015": Object {
      "id": "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
      "name": "Meat",
    },
  },
}
`;

exports[`categories reducer addCategory should add a category to the list with an id 1`] = `
Object {
  "allIds": Array [
    "meat",
  ],
  "byId": Object {
    "meat": Object {
      "id": "meat",
      "name": "Meat",
    },
  },
}
`;

exports[`categories reducer can be used with reduce 1`] = `
Object {
  "allIds": Array [
    "dairy",
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
  ],
  "byId": Object {
    "c90399f41b3fd3c0af1e10c70e03eb5369d79015": Object {
      "id": "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
      "name": "Meat",
    },
    "dairy": Object {
      "id": "dairy",
      "name": "Milk & Cheese",
    },
  },
}
`;

exports[`categories reducer nameCategory should set the name of the selected category from the list 1`] = `
Object {
  "allIds": Array [
    "dairy",
    "meat",
  ],
  "byId": Object {
    "dairy": Object {
      "name": "Milk & Cheese",
    },
    "meat": Object {
      "name": "Meat",
    },
  },
}
`;

exports[`categories reducer removeCategory should remove a category from the list 1`] = `
Object {
  "allIds": Array [
    "meat",
  ],
  "byId": Object {
    "meat": Object {
      "name": "Meat",
    },
  },
}
`;
